### PR TITLE
support ingest external file

### PR DESCRIPTION
**Pull request by hhkbp2**
Hi,

This PR try to take over PR #54

### CODE REVIEW

**Comment by hhkbp2** [comment]
PTAL @zhangjinpeng1987 @siddontang @BusyJay

File: librocksdb_sys/src/lib.rs [hunk 1]
```diff
@@ -490,6 +493,54 @@ extern "C" {
                                                           ignore_snapshot: bool);
     pub fn crocksdb_compactionfilter_destroy(filter: *mut DBCompactionFilter);
 
+    // EnvOptions
+    pub fn rocksdb_envoptions_create() -> *mut EnvOptions;
```
Comment (siddontang) [inline librocksdb_sys/src/lib.rs]: please add the C API with the prefix `crocksdb` in c.h, c.cc.

File: src/rocksdb_options.rs [hunk 2]
```diff
@@ -620,6 +620,87 @@ impl Drop for FlushOptions {
     }
 }
 
+/// IngestExternalFileOptions is used by DB::ingest_external_file
+pub struct IngestExternalFileOptions {
+    pub inner: *mut crocksdb_ffi::IngestExternalFileOptions,
+}
+
+impl IngestExternalFileOptions {
```
Comment (siddontang) [inline src/rocksdb_options.rs]: do we cover all the IngestExternalFileOptions in test?

File: librocksdb_sys/src/lib.rs [hunk 3]
```diff
@@ -616,4 +667,104 @@ mod test {
...
         }
     }
+
+    unsafe fn check_get(db: *mut DBInstance,
+                        opt: *const DBReadOptions,
+                        key: &[u8],
+                        val: Option<&[u8]>) {
+        let mut val_len = 0;
+        let mut err = ptr::null_mut();
+        let res_ptr = rocksdb_get(db, opt, key.as_ptr(), key.len(), &mut val_len, &mut err);
```
Comment (siddontang) [inline librocksdb_sys/src/lib.rs]: please use crocksdb in all place.

File: src/rocksdb_options.rs (hunk 2 above)
Comment (hhkbp2) [inline src/rocksdb_options.rs]: The test case in `test_ingest_external_file.rs` test the behavior with the option `move_files` enabled and disabled, with other options in default value, which is enough for our usage case.

**Comment by hhkbp2** [comment]
PTAL @zhangjinpeng1987 @siddontang @BusyJay

**Comment by siddontang** [comment]
LGTM
PTAL @BusyJay

**Comment by zhangjinpeng1987** [comment]
LGTM

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+all: build
+
+dev: build test
+
+build:
+	$(QUIET) cargo build
+
+test:
+	$(QUIET) cargo test
```
Comment (BusyJay) [inline Makefile]: Missing `&&`

File: librocksdb_sys/build.rs [hunk 5]
```diff
@@ -13,6 +13,7 @@ macro_rules! t {
 }
 
 fn main() {
+    println!("rerun-if-changed=build.sh");
```
Comment (BusyJay) [inline librocksdb_sys/build.rs]: Not necessary.

File: Makefile (hunk 4 above)
Comment (hhkbp2) [inline Makefile]: If the command in line 18 fails, it will stop running the below commands. This is the same as `&&`.

**Comment by hhkbp2** [comment]
PTAL @BusyJay

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+	$(QUIET) cd $(lib_rocksdb_dir) && cargo test
```
Comment (BusyJay) [inline Makefile]: This doesn't seem to work on Linux.

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**